internal/controllers/image/reconcile.go

```diff
@@ -103,59 +103,58 @@ func (r *orcImageReconciler) reconcileNormal(ctx context.Context, orcImage *orcv
-				// An image we previously referenced has been deleted unexpectedly. We can't recover from this.
-				err = capoerrors.Terminal(orcv1alpha1.OpenStackConditionReasonUnrecoverableError, "image has been deleted from glance")
-			}
-			return image, err
-		}
-
-		if orcImage.Spec.Import != nil {
-			log.V(4).Info("Importing existing glance image")
-
-			if orcImage.Spec.Import.ID != nil {
-				image, err := imageClient.GetImage(*orcImage.Spec.Import.ID)
-				if capoerrors.IsNotFound(err) {
-					// We assume that an image imported by ID must already exist. It's a terminal error if it doesn't.
-					err = capoerrors.Terminal(orcv1alpha1.OpenStackConditionReasonUnrecoverableError, "referenced image does not exist in glance")
-				}
-				return image, err
-			}
-
-			listOpts := listOptsFromImportFilter(orcImage.Spec.Import.Filter)
-			return getGlanceImageFromList(ctx, listOpts, imageClient)
+	var glanceImage *images.Image
+	switch {
+	case orcImage.Status.ID != nil:
+		log.V(4).Info("Fetching existing glance image", "ID", *orcImage.Status.ID)
 
-			// TODO: When we support 'import and manage' we need to implement
-			// setting spec.resource from the discovered glance image here.
+		glanceImage, err = imageClient.GetImage(*orcImage.Status.ID)
+		if capoerrors.IsNotFound(err) {
+			// An image we previously referenced has been deleted unexpectedly. We can't recover from this.
+			err = capoerrors.Terminal(orcv1alpha1.OpenStackConditionReasonUnrecoverableError, "image has been deleted from glance")
+		}
+		if err != nil {
+			return ctrl.Result{}, err
 		}
 
-		log.V(4).Info("Checking for previously created image")
-
-		listOpts := listOptsFromCreation(orcImage)
-		return getGlanceImageFromList(ctx, listOpts, imageClient)
-	}()
-	if err != nil {
-		return ctrl.Result{}, err
-	}
+	case orcImage.Spec.Import.ID != nil:
+		log.V(4).Info("Importing existing Glance image by ID")
+		glanceImage, err = imageClient.GetImage(*orcImage.Status.ID)
+		if capoerrors.IsNotFound(err) {
+			// An image we previously referenced has been deleted unexpectedly. We can't recover from this.
+			err = capoerrors.Terminal(orcv1alpha1.OpenStackConditionReasonUnrecoverableError, "image has been deleted from glance")
+		}
+		if err != nil {
+			return ctrl.Result{}, err
+		}
 
-	if glanceImage == nil {
-		if orcImage.Spec.Import != nil {
+	case orcImage.Spec.Import.Filter != nil:
+		log.V(4).Info("Importing existing Glance image by filter")
+		listOpts := listOptsFromImportFilter(orcImage.Spec.Import.Filter)
+		glanceImage, err = getGlanceImageFromList(ctx, listOpts, imageClient)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if glanceImage == nil {
 			log.V(3).Info("Image does not yet exist")
 			addStatus(withProgressMessage("Waiting for glance image to be created externally"))
-
 			return ctrl.Result{RequeueAfter: waitForGlanceImageStatusUpdate}, err
 		}
 
-		glanceImage, err = createImage(ctx, orcImage, imageClient)
+	default:
```

Comment:
Lets have a comment here about how this is the 'adopt' branch for a resource we created by failed to record.

Reply:
I have re-added the log line, which I had removed by mistake. Does that count as a comment?